
<launch>
  <!-- arguments for camera input -->
  <arg name="filter_size" default="0.005"/>
  <arg name="x_min" default="-5"/>
  <arg name="x_max" default="5"/>
  <arg name="y_min" default="-5"/>
  <arg name="y_max" default="5"/>
  <arg name="z_min" default="-5"/>
  <arg name="z_max" default="5"/>
  
  <arg name="camera_map" default="$(find azure_kinect_manager)/maps/camera"/>
  <arg name="world_map" default="$(find azure_kinect_manager)/maps/world/map.yaml"/>

  <!-- Start camera broadcaster-->
  <node name="single_azure_manager" pkg="azure_kinect_manager" 
        type="azure_manager.py" respawn="false" output="screen" >
      <param name="camera_name" type="string" value="azure1" />
      <param name="camera_map" value="$(arg camera_map)" />
      <param name="world_map" value="$(arg world_map)" />
      <param name="filter_size" value="$(arg filter_size)" />
      <param name="x_min" value="$(arg x_min)" />
      <param name="x_max" value="$(arg x_max)" />
      <param name="y_min" value="$(arg y_min)" />
      <param name="y_max" value="$(arg y_max)" />
      <param name="z_min" value="$(arg z_min)" />
      <param name="z_max" value="$(arg z_max)" />
  </node>

  <node pkg="tf" type="static_transform_publisher" 
        name="azure_base_broadcaster" args="0 0 0 0 0 0 world map 100" /> 

</launch>